### ISSUE TITLE
[DX] Revamp all the documentation

### ISSUE THREAD

**Issue by javiereguiluz** [FriendsOfSymfony/FOSUserBundle#1577]
I'd love to revamp the documentation of this bundle, which is used by more than 60% of all Symfony developers and it was voted as the most useful Symfony plugin.

But before starting this huge task, I'd like to know if the bundle creators and/or managers are willing to include major changes in the documentation.

This is a short overview of the changes that I'm considering:
- Better explain the purpose of the bundle. This is **not** just to store users in the database.
- Clearly separate the outdated Symfony 2.0 documentation.
- Improve the [Step 3 of the installation](https://github.com/FriendsOfSymfony/FOSUserBundle/blob/master/Resources/doc/index.md#step-3-create-your-user-class) which currently is a mess.
- [Logging by username or email](https://github.com/FriendsOfSymfony/FOSUserBundle/blob/master/Resources/doc/logging_by_username_or_email.md) should not be a separate document.
- Better promote the articles useful for most developers: How to send registration confirmation emails, How to enable password reset emails, How useful is to use the User Manager, How to set up an invitation-only application, etc.
- Millions of minor tweaks and simplifications.

---

Documentation update roadmap:
- [x] Remove any reference about Symfony 2.0.
- [x] Switch to RST formatting instead of using MD (first convert the 1.3 version and then the master version).
- [ ] Better explain the purpose of the plugin.
- [ ] Simplify the explanation of the Step 3 of the installation process.
- [ ] Merge some of the articles into the main "tutorial" of the bundle (e.g. Logging by username or email)
- [ ] Promote more prominently the articles about the common problems that this bundle solves: 1) send registration emails; 2) send "I forgot my password" emails, ~~3) send codes to invitation-only apps.~~
- [ ] ...

**Comment by javiereguiluz** [comment]
I'm closing this 4-year old issue because I won't be able to work on this beyond the two completed tasks. If anyone volunteers to do the remaining tasks, please create a new issue explaining what do you want to do to improve docs and wait for approval from the maintainers of this bundle. Thanks!
